Application/EdFi.Ods.Features/OpenApiMetadata/Providers/OpenApiV3UpconversionProvider.cs

```diff
@@ -0,0 +1,110 @@
+        _httpContextAccessor = httpContextAccessor;
+        _tenantsConfigurationOptions = tenantsConfigurationOptions;
+    }
+
+    public string GetUpconvertedOpenApiJson(string openApiJson)
+    {
+        var openApiDocument = new OpenApiStringReader().Read(openApiJson, out _);
+        PopulateServersConfiguration(ref openApiDocument);
+        return openApiDocument.SerializeAsJson(OpenApiSpecVersion.OpenApi3_0);
+    }
+
+    private void PopulateServersConfiguration(ref OpenApiDocument openApiDocument)
+    {
+        openApiDocument.Servers.Clear();
+
+        var uriBase =
+            $"{_httpContextAccessor.HttpContext?.Request.Scheme(this._reverseProxySettings)}://{_httpContextAccessor.HttpContext?.Request.Host(this._reverseProxySettings)}:{_httpContextAccessor.HttpContext?.Request.Port(this._reverseProxySettings)}/";
+
+        var routeContextSegment = "";
+
+        if (!string.IsNullOrEmpty(_apiSettings.OdsContextRouteTemplate))
+        {
+            routeContextSegment = $"/{{{string.Join("}/{", _apiSettings.GetOdsContextRouteTemplateKeys())}}}";
+        }
+
+        var odsServer = new OpenApiServer()
+        {
+            Url = $"{uriBase}{{currentTenant}}{routeContextSegment}{{apiDataPath}}",
+            Variables = new Dictionary<string, OpenApiServerVariable>()
+        };
+
+        openApiDocument.Servers.Add(odsServer);
+
+        try
+        {
+            foreach (var server in openApiDocument.Servers)
+            {
+                if (!string.IsNullOrEmpty(_apiSettings.OdsContextRouteTemplate))
+                {
+                    foreach (var contextRouteTemplateKey in _apiSettings.GetOdsContextRouteTemplateKeys())
+                    {
+                        var routeContextValues =
+                            _IEdFiAdminRawOdsInstanceConfigurationDataProvider.GetDistinctOdsInstanceContextValuesAsync(
+                                contextRouteTemplateKey);
+
+                        server.Variables.Add(
+                            contextRouteTemplateKey,
+                            new OpenApiServerVariable()
+                            {
+                                Enum = routeContextValues
+                                    .Result.ToList(),
+                                Default = routeContextValues
+                                    .Result.ToList().Last()
+                            });
+
+                        openApiDocument.Components.SecuritySchemes.Single().Value.Flows.ClientCredentials.TokenUrl = new Uri(
+                            $"{uriBase}/{routeContextValues.Result.ToList().Last()}{{currentTenant}}oauth/token");
+                    }
+                }
+                else
+                {
+                    openApiDocument.Components.SecuritySchemes.Single().Value.Flows.ClientCredentials.TokenUrl = new Uri(
+                        $"{uriBase}{{currentTenant}}oauth/token");
+                }
+            }
+        }
+        catch { }

```

Comment:
This catch is hiding the exception thrown because the sql function is not present in Postgresql